src/page-test/TestPage.tsx

```diff
@@ -83,6 +86,25 @@ const TestPage: React.FC = () => {
         <div className="ic-facts">ic-facts</div>
         <div className="ic-share">ic-share</div>
       </div>
+
+      <div>
+        <button
+          onClick={() =>
+            setValues({ ...values, val1: (values.val1 as number) + 1 })
+          }
+        >
+          Val1: {values.val1}
+        </button>
+        <button
+          onClick={() =>
+            setValues({ ...values, val2: (values.val2 as number) + 1 })
+          }
+        >
+          Val2: {values.val2}
+        </button>
+        <button onClick={() => setAction("push")}>Push History</button>
+        <button onClick={() => setAction("replace")}>Replace History</button>
```

Comment:
On the form, _every_ form value update should replace history, not just when the user clicks a dedicated button. (aka, we'd need to call `setValues(...)`, then `setAction("replace")`, then `setAction("push")` some time later instead of just each `setValues(...)` above.)

Reply:
Yeah, the default behavior is that every form value update replaces the history. The buttons here merely provide an explicit way of testing the hook behavior.

Reply:
Originally designed that way because we'd be `replacing` more than we'd be `pushing` (from our convo yesterday)